collector/recon.go

```diff
@@ -0,0 +1,681 @@
+	exitCode := 0
+	cmdArgs := []string{"--unmounted", "--verbose"}
+	outputPerHost, err := getSwiftReconOutputPerHost(t.pathToReconExecutable, cmdArgs...)
+	if err == nil {
+		for hostname, dataBytes := range outputPerHost {
+			var disksData []struct {
+				Device string `json:"device"`
+			}
+			err := json.Unmarshal(dataBytes, &disksData)
+			if err != nil {
+				exitCode = 1
+				logg.Error("swift dispersion: %s: %s: %s", cmdArgsToStr(cmdArgs), hostname, err.Error())
+				continue // to next host
+			}
+
+			ch <- t.unmountedDrives.mustNewConstMetric(float64(len(disksData)), hostname)
+		}
+	} else {
+		exitCode = 1
+		logg.Error("swift recon: %s: %s", cmdArgsToStr(cmdArgs), err.Error())
+	}
+
+	ch <- exitCodeTypedDesc.mustNewConstMetric(float64(exitCode), cmdArgsToStr(cmdArgs))
+}
+
+// reconDriveAuditTask implements the collector.collectorTask interface.
+type reconDriveAuditTask struct {
+	pathToReconExecutable string
+	auditErrors           typedDesc
+}
+
+func newReconDriveAuditTask(pathToReconExecutable string) collectorTask {
+	return &reconDriveAuditTask{
+		pathToReconExecutable: pathToReconExecutable,
+		auditErrors: typedDesc{
+			desc: prometheus.NewDesc(
+				"swift_cluster_drives_audit_errors",
+				"Drive audit errors reported by the swift-recon tool.",
+				[]string{"storage_ip"}, nil),
+			valueType: prometheus.GaugeValue,
+		},
+	}
+}
+
+// reconDriveAuditTask implements the collector.collectorTask interface.
+func (t *reconDriveAuditTask) describeMetrics(ch chan<- *prometheus.Desc) {
+	t.auditErrors.describe(ch)
+}
+
+// reconDriveAuditTask implements the collector.collectorTask interface.
+func (t *reconDriveAuditTask) collectMetrics(ch chan<- prometheus.Metric, exitCodeTypedDesc typedDesc) {
+	exitCode := 0
+	cmdArgs := []string{"--driveaudit", "--verbose"}
+	outputPerHost, err := getSwiftReconOutputPerHost(t.pathToReconExecutable, cmdArgs...)
+	if err == nil {
+		for hostname, dataBytes := range outputPerHost {
+			var data struct {
+				DriveAuditErrors int64 `json:"drive_audit_errors"`
+			}
+			err := json.Unmarshal(dataBytes, &data)
+			if err != nil {
+				exitCode = 1
+				logg.Error("swift dispersion: %s: %s: %s", cmdArgsToStr(cmdArgs), hostname, err.Error())
+				continue // to next host
+			}
+
+			ch <- t.auditErrors.mustNewConstMetric(float64(data.DriveAuditErrors), hostname)
+		}
+	} else {
+		exitCode = 1
+		logg.Error("swift recon: %s: %s", cmdArgsToStr(cmdArgs), err.Error())
+	}
+
+	ch <- exitCodeTypedDesc.mustNewConstMetric(float64(exitCode), cmdArgsToStr(cmdArgs))
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Helper functions.
+
+var reconHostOutputRx = regexp.MustCompile(`(?m)^-> https?://([a-zA-Z0-9-.]+)\S*\s(.*)$`)
+
+func getSwiftReconOutputPerHost(pathToExecutable string, cmdArgs ...string) (map[string][]byte, error) {
+	cmd := exec.Command(pathToExecutable, cmdArgs...)
+	out, err := cmd.Output()
```

Comment:
```suggestion
	out, err := cmd.CombinedOutput()
```
so that we also see logs from stderr in the error message.